prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -904,15 +904,8 @@ class ActionHistory {
                     }
                 }
             }
-            if (msg.isNotEmpty()) {
-                if (failOnError) {
-                    error("*** Sanity check comparing old Headers list to new ReportFile list FAILED:  $msg")
-                } else {
-                    println(
-                        "************ FAILURE: sanity check comparing old Headers " +
-                            "list to new ReportFiles list FAILED:  $msg\""
-                    )
-                }
+            if (msg.isNotEmpty() && failOnError) {
```

Comment:
Should there still be somewhere to print error messages if `failOnError = false`?

Reply:
I would change the print to logger.error.

Reply:
@carlosfelix2 Unfortunately there's no `logger` (or `context.logger`) in scope here -- this is a companion object. But my kotlin is super shallow -- if there's a way to reference one elsewhere I'd love a pointer.

Reply:
...also, I figured making these changes (in ActionHistory) wouldn't be too big a deal because both of these functions (`sanityCheckReport` and `sanityCheckReports`) have a "Get rid of this..." comment.

Reply:
fixed

Reply:
I think all you need to do is add ": Logging" to the companion object.